src/component.ts

```diff
@@ -103,7 +97,26 @@ export function extendWithHostListenersAndChildren(ctrl: {new(...args: any[])},
         if (viewChildEls.length) {
           this[property] = child.first ? viewChildEls[0] : viewChildEls;
         }
+        else {
```

Comment:
please remove else condition, I am not sure it's a good idea to remove user defined property from parent class, better leave it, so it be undefined instead.
Unless you have any other rational for adding this else condition

Reply:
Because if the element referenced is removed from the DOM, then you need to remove the reference or it won't be garbage collected. But is it supposed to be done only when the parent is removed? I'm not sure what's the Angular logic on this

Reply:
Oh sorry, actually this is only necessary when binding to a single element, it should probably be:
```
else if (child.first) {
  this[property] = undefined;
}
```
but without this you would keep a reference to the controller of an element that does not exist anymore

Reply:
Ah, right, so it's better to set to undefined I think

Reply:
Done